Shared/Samples/Measure distance in scene/MeasureDistanceInSceneView.swift

```diff
@@ -0,0 +1,175 @@
+// Copyright 2023 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+import ArcGIS
+
+struct MeasureDistanceInSceneView: View {
+    /// The view model for the sample.
+    @StateObject private var model = Model()
+    
+    /// A string for the direct distance of the location distance measurement.
+    @State private var directDistanceText = "--"
+    
+    /// A string for the horizontal distance of the location distance measurement.
+    @State private var horizontalDistanceText = "--"
+    
+    /// A string for the vertical distance of the location distance measurement.
+    @State private var verticalDistanceText = "--"
+    
+    /// The unit system for the location distance measurement, selected by the picker.
+    @State private var unitSystemSelection: UnitSystem = .metric
+    
+    /// The overlay instruction message text.
+    @State private var instructionText: String = .startMessage
+    
+    var body: some View {
+        VStack {
+            SceneViewReader { sceneViewProxy in
+                SceneView(scene: model.scene, analysisOverlays: [model.analysisOverlay])
+                    .onSingleTapGesture { screenPoint, _ in
+                        // Set the start and end locations when the screen is tapped.
+                        Task {
+                            guard let location = try? await sceneViewProxy.location(
+                                fromScreenPoint: screenPoint
+                            ) else { return }
+                            
+                            if model.locationDistanceMeasurement.startLocation != model.locationDistanceMeasurement.endLocation {
+                                model.locationDistanceMeasurement.startLocation = location
+                                instructionText = .endMessage
+                            } else {
+                                instructionText = .startMessage
+                            }
+                            model.locationDistanceMeasurement.endLocation = location
+                        }
+                    }
+                    .onDragGesture { _, _ in
+                        if model.locationDistanceMeasurement.startLocation == model.locationDistanceMeasurement.endLocation {
+                            return true
+                        }
+                        return false
```

Comment:
```suggestion
                        if model.locationDistanceMeasurement.startLocation == model.locationDistanceMeasurement.endLocation {
                            return true
                        } else {
                            return false
                        }
```

Reply:
Or just `return model.locationDistanceMeasurement.startLocation == model.locationDistanceMeasurement.endLocation`.

Reply:
Phil's suggestion is good, let's apply that and add a comment to explain when the drag gesture is active.